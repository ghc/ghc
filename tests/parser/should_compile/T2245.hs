default (T)

data T = T deriving (Eq, Ord, Read, Show)
instance Num T
instance Fractional T

main = interact $ show . (< 1e400) . read