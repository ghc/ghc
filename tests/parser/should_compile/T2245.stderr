T2245.hs:4:10: warning: [GHC-06201] [-Wmissing-methods (in -Wdefault)]
    • No explicit implementation for
        ‘+’, ‘*’, ‘abs’, ‘signum’, ‘fromInteger’, and (either ‘negate’
                                                              or
                                                              ‘-’)
    • In the instance declaration for ‘Num T’

T2245.hs:5:10: warning: [GHC-06201] [-Wmissing-methods (in -Wdefault)]
    • No explicit implementation for
        ‘fromRational’ and (either ‘recip’ or ‘/’)
    • In the instance declaration for ‘Fractional T’

T2245.hs:7:29: warning: [GHC-18042] [-Wtype-defaults (in -Wall)]
    • Defaulting the type variable ‘b0’ to type ‘T’ in the following constraints
        (Fractional b0)
          arising from the literal ‘1e400’ at T2245.hs:7:29-33
        (Ord b0) arising from a use of ‘<’ at T2245.hs:7:27
        (Read b0) arising from a use of ‘read’ at T2245.hs:7:38-41
    • In the expression: 1e400
      In the first argument of ‘(.)’, namely ‘(< 1e400)’
      In the second argument of ‘(.)’, namely ‘(< 1e400) . read’

